server/src/graql/gremlin/TraversalPlanner.java

```diff
@@ -228,6 +240,68 @@ public static GraqlTraversal createTraversal(Pattern pattern, TransactionOLTP tx
         return startingNodes;
     }
 
+    /**
+     * Estimate the "cost" of a starting point for each type of fixed cost fragment
+     * These are heuristic proxies using statistics
+     * @param fixedCostFragment
+     * @return
+     */
+    private static long estimateInitialCost(Fragment fixedCostFragment, TransactionOLTP tx) {
+        KeyspaceStatistics statistics = tx.session().keyspaceStatistics();
+        if (fixedCostFragment instanceof  LabelFragment) {
```

Comment:
Maybe let's put all this instance specific behaviours into respective classes. I know we want to do the big refactor in the end, but it should be still beneficial to avoid cluttering things now if we can.

Reply:
great plans